base/errorshow.jl

```diff
@@ -1086,6 +1092,31 @@ end
 
 Experimental.register_error_hint(methods_on_iterable, MethodError)
 
+# Display a hint in case the user tries to access non-member fields of container type datastructures
+function fielderror_hint_handler(io, exc)
+    @nospecialize
+    field = exc.field
+    type = exc.type
+    if type <: Dict
+        println(io,
+            """
+            \nDid you mean to access dict values using key: `:$field` ?
+            Consider using indexing syntax.
+
+            # Example
+            julia> dict = Dict(:$(field)=>5)
```

Comment:
Seems odd to use markdown syntax here since this will just get printed to stderr.

Reply:
Also, the error hints tend to be quite short and to the point. Something like:

> Dictionaries are indexed using the [key] syntax 

or something along those lines should be enough.

Reply:
For most markdown syntax (e.g. code formatting), I agree with you (see above conversation in this PR). However, the single # header is also reasonable plaintext "syntax".

This is a very niche error case (it only happens when you call `my_dict.foo`) for some `my_dict::Dict`, so it is unlikely that this will come across as spammy. The only case I can think of where this extended explanation would be unwelcome is if folks are intentionally attempting to access internal Dict fields and also making typos.

Reply:
It's good to keep the styling consistent with the rest of the hints for things to be cohesive. Following the example of `*` on `AbstractString` here would be the simplest.